src/jvm/main/org/jetbrains/kotlinx/lincheck/execution/ExecutionGenerator.kt

```diff
@@ -0,0 +1,44 @@
+/*
+* #%L
+* Lincheck
+* %%
+* Copyright (C) 2015 - 2018 Devexperts, LLC
+* %%
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Lesser Public License for more details.
+*
+* You should have received a copy of the GNU General Lesser Public
+* License along with this program.  If not, see
+* <http://www.gnu.org/licenses/lgpl-3.0.html>.
+* #L%
+*/
+package org.jetbrains.kotlinx.lincheck.execution
+
+import org.jetbrains.kotlinx.lincheck.CTestConfiguration
+import org.jetbrains.kotlinx.lincheck.CTestStructure
+
+/**
+ * Implementation of this interface generates execution scenarios.
+ * By default, [RandomExecutionGenerator] is used.
+ *
+ *
+ * IMPORTANT!
+ * All implementations should have the same constructor as [ExecutionGenerator] has.
+ */
+abstract class ExecutionGenerator protected constructor(protected val testConfiguration: CTestConfiguration, protected val testStructure: CTestStructure) {
```

Comment:
Please remove `protected constructor`. It is also better to put each argument on a separate line.